src/pygeometadata/mcf.py

```diff
@@ -0,0 +1,413 @@
+                }
+            else:
+                template[prop] = get_template(sch)
+        return template
+
+    elif 'type' in schema and schema['type'] == 'array':
+        return [get_template(schema['items'])]
+    else:
+        return get_default(schema)
+
+
+class MCF:
+    """Encapsulates the Metadata Control File and methods for populating it.
+
+    A Metadata Control File (MCF) is a YAML file that complies with the
+    MCF specification defined by pygeometa.
+    https://github.com/geopython/pygeometa
+
+    Attributes:
+        datasource (string): path to dataset to which the metadata applies
+        mcf (dict): dict representation of the MCF
+
+    """
+
+    def __init__(self, source_dataset_path):
+        """Create an MCF instance, populated with inherent values.
+
+        The MCF will be valid according to the pygeometa schema. It has
+        all required properties. Instrinsic properties of the dataset
+        are used to populate as many MCF properties as possible. And
+        default/placeholder values are used for properties that require
+        user input.
+
+        Args:
+            source_dataset_path (string): path to dataset to which the metadata
+                applies
+
+        """
+        self.datasource = source_dataset_path
+        self.mcf_path = f'{self.datasource}.yml'
+        self.mcf = None
+
+        if os.path.exists(self.mcf_path):
+            try:
+                # pygeometa.core.read_mcf can parse nested MCF documents,
+                # where one MCF refers to another
+                self.mcf = pygeometa.core.read_mcf(self.mcf_path)
+                self.validate()
+            except (pygeometa.core.MCFReadError, ValidationError) as err:
+                LOGGER.warning(err)
+                self.mcf = None
+
+        if self.mcf is None:
+            self.mcf = get_template(MCF_SCHEMA)
+            self.mcf['mcf']['version'] = \
+                MCF_SCHEMA['properties']['mcf']['properties']['version']['const']
+            # fill all values that can be derived from the dataset
+            self.get_spatial_info()
+
+    def add_metadata_attr(self, attribute):
+        """Add an arbitrary attribute to the metadata.
+
+        These should be attributes that do not appear elsewhere in the MCF
+        specification.
+
+        Args:
+            attribute (dict)
+
+        """
+        if 'attributes' not in self.mcf:
+            self.mcf['attributes'] = []
+        self.mcf['attributes'].append(attribute)
+
+    def title(self, title):
+        """Add a title for the dataset.
+
+        Args:
+            title (str)
+
+        """
+        self.mcf['identification']['title'] = title
+
+    def abstract(self, abstract):
+        """Add an abstract for the dataset.
+
+        Args:
+            abstract (str)
+
+        """
+        self.mcf['identification']['abstract'] = abstract
+
+    def keywords(self, keywords, section='default', keywords_type='theme',
```

Comment:
Oh that's interesting. One concern is that our object is like,
```python
mcf = MCF(...)
print(mcf.mcf)  # the actual metadata dictionary
```
Since an instance of `MCF` has a couple other attributes besides the metadata dict. I wonder if there is a better way to design that, though. Would it be better if an instance of `MCF` was a direct representation of the MCF file?

Good advice on changing the method names, I'll do that.

Reply:
>Since an instance of MCF has a couple other attributes besides the metadata dict. I wonder if there is a better way to design that, though. Would it be better if an instance of MCF was a direct representation of the MCF yaml?

Yep, I hear you on not wanting to allow access to these other attributes!  The usual python approach would be to make anything you don't want a user to access "private" by adding a `_` prefix.  [Here's a neat SO discussion](https://stackoverflow.com/a/7456865) about this in python and how attribute access compares to other languages.

>Would it be better if an instance of MCF was a direct representation of the MCF yaml?
As you said, I think the complex nested structure of the MCF yaml means that if we exclusively use that structure, then users of this library will need to be aware of the structure in order to use this library, which we don't want.  An interested informed user can always work directly with the upstream `pygeometa`, so I think we should be free to do what we think is right to make it easier to use and write spatial metadata.